finagle-http/src/main/scala/com/twitter/finagle/http/filter/ComplianceFilter.scala

```diff
@@ -0,0 +1,62 @@
+package com.twitter.finagle.http.filter
+
+import com.twitter.finagle.http.{Fields, Request, Response, Status}
+import com.twitter.finagle._
+import com.twitter.logging.Logger
+import com.twitter.util.Future
+import org.jboss.netty.buffer.ChannelBuffers
+
+/**
+ * Modify a content and header field filter.
+ *
+ * @see [[https://tools.ietf.org/html/rfc2616 "Hypertext Transfer Protocol -- HTTP/1.1"]]
+ * @see [[https://tools.ietf.org/html/rfc7230 "Hypertext Transfer Protocol (HTTP/1.1): Message Syntax and Routing"]]
+ * section-3.3.2 on `Content-Length` for more information.
+ */
+private[finagle] class ComplianceFilter[Req <: Request] extends SimpleFilter[Req, Response] {
+  import ComplianceFilter._
+  val logger = Logger(getClass)
+
+  def apply(request: Req, service: Service[Req, Response]): Future[Response] = {
+    service(request).map { rep =>
+      if (mustNotIncludeContent(rep.status)) {
+        if (rep.getContent() != ChannelBuffers.EMPTY_BUFFER) {
+          rep.clearContent()
+          logger.warning(s"Response with a status code of ${rep.statusCode} must not have a body-message thus the content has been removed.")
+        }
+
+        if (rep.status == Status.NoContent && rep.contentLength.isDefined) {
+          rep.headerMap.remove(Fields.ContentLength)
+          logger.warning(s"Response with a status code of ${rep.statusCode} must not have a Content-Length header field thus the field has been removed.")
+        }
+
+        //Make sure a Content-Length header field is set with a value of 0 if it is explicitly sent for a 304 response
```

Comment:
> I guess this is something that users must set?

I think so if users really want to send the header field in a 304 response... I'm not sure how to _validate_ that they've chosen the _correct_ Content-Length value. As the RFC says, if the field-value doesn't equal the decimal number of octets that would have been sent in the payload body of a 200 (OK) response to the same request then we MUST NOT send Content-Length.

> I guess it's better to just not modify the content length in this case.

Yeah, not modifying is better I suppose so. Will fix it :)

I feel that things will be more complicated if we start to think about a Content-Length value of a 304 response as I've mentioned. Do you think we could stick to the original problems we had before and hopefully we can find a way to handle the new issue we've found now? What do you think?